pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -234,6 +289,11 @@ private List<Object[]> buildJoinedDataBlockAnti(MseBlock.Data leftBlock) {
     for (Object[] leftRow : leftRows) {
       Object key = _leftKeySelector.getKey(leftRow);
       // ANTI-JOIN only checks non-existence of the key
+      // For ANTI-JOIN, rows with null keys should be included (null != anything)
+      if (isNullKey(key)) {
+        rows.add(leftRow);
+        continue;
+      }
```

Comment:
Hm, I'm not sure this is the right thing to do. Taking Postgres for instance (which is usually the reference we use in Pinot for SQL semantics):

```
postgres=# CREATE TABLE t1 (i int);
CREATE TABLE
postgres=# CREATE TABLE t2 (i int);
CREATE TABLE
postgres=# INSERT INTO t1 VALUES (1), (2), (null);
INSERT 0 3
postgres=# INSERT INTO t2 VALUES (1), (2), (null);
INSERT 0 3

postgres=# SELECT * FROM t2 where i not in (SELECT i FROM t2);
 i
---
(0 rows)

```